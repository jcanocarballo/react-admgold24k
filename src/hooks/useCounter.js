import { useState } from "react"

export const useCounter = (initialState = 10) => {
  const [counter, setCouter] = useState(initialState);

  const increment = () => {
    setCouter(counter + 1)
  }

  const decrement = () => {
    setCouter(counter - 1)
  }

  const reset = () => {
    setCouter(initialState)
  }

  return {
    counter,
    increment,
    decrement,
    reset
  }
}
