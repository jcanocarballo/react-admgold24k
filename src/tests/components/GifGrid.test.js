const { GifGrid } = require("../../components/GifGrid");
import {shallow} from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas del archivo GifGrid', () => {
  let category = 'hola mundo';
  // let wraper = shallow(<GifGrid category={category}></GifGrid>);
  // beforeEach( () => {
  //   wraper = shallow(<GifGrid category={category}></GifGrid>);
  // });
  test('ebe mostrar el componente <GifGrid /> correctamente ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: false
    });
    const wraper = shallow(<GifGrid category={category}></GifGrid>);
    expect(wraper).toMatchSnapshot();
  });

  test('debe mostar items cuando se cargan imagenes UseFechGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/imagen.png',
        title: 'un titulo'
      }
    ]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wraper = shallow(<GifGrid category={category}></GifGrid>);
    //expect(wraper).toMatchSnapshot();
    expect(wraper.find('p').exists()).toBe(false);
    expect(wraper.find('GifGridItem').length).toBe(gifs.length);
  });
});