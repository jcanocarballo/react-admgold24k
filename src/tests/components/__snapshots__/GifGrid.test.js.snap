// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo GifGrid ebe mostrar el componente <GifGrid /> correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    hola mundo
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
